File: flutter/lib/screens/appointment_detail_screen.dart
import 'package:flutter/material.dart';

import '../models/appointment.dart';
import '../theme/app_colors.dart';
import '../utils/appointment_date_format.dart';
import '../widgets/ui_components.dart';

class AppointmentDetailScreen extends StatelessWidget {
  const AppointmentDetailScreen({super.key, required this.appointment});

  final Appointment appointment;

  @override
  Widget build(BuildContext context) => Scaffold(
    body: AppPage(
      child: Column(
        children: [
          Container(
            width: double.infinity,
            padding: const EdgeInsets.fromLTRB(10, 25, 18, 13),
            decoration: const BoxDecoration(
              border: Border(bottom: BorderSide(color: Color(0xFFD9E2EA))),
            ),
            child: Row(
              children: [
                IconButton(
                  tooltip: 'Back to appointments',
                  constraints: const BoxConstraints(
                    minWidth: 48,
                    minHeight: 48,
                  ),
                  icon: const Icon(
                    Icons.arrow_back_ios_new,
                    color: AppColors.primary,
                  ),
                  onPressed: () => Navigator.of(context).pop(),
                ),
                const SizedBox(width: 8),
                Expanded(
                  child: Text(
                    'Appointment Details',
                    style: Theme.of(context).textTheme.headlineMedium,
                  ),
                ),
              ],
            ),
          ),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.fromLTRB(18, 20, 18, 28),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Semantics(
                    container: true,
                    label:
                        '${appointment.status.label}, ${appointment.visitTitle}, ${appointment.clinicianName}',
                    child: ExcludeSemantics(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          AppointmentStatusPill(status: appointment.status),
                          const SizedBox(height: 17),
                          Text(
                            appointment.visitTitle,
                            style: Theme.of(context).textTheme.headlineMedium,
                          ),
                          const SizedBox(height: 9),
                          Text(
                            appointment.clinicianName,
                            style: Theme.of(context).textTheme.titleLarge,
                          ),
                        ],
                      ),
                    ),
                  ),
                  const SizedBox(height: 29),
                  _DetailSection(
                    label: 'Date & time',
                    value: appointmentDetailLabel(appointment.scheduledAt),
                  ),
                  _DetailSection(
                    label: 'Location',
                    value: appointment.locationLabel,
                  ),
                  _DetailSection(
                    label: 'Visit type',
                    value: appointment.visitFormat.label,
                    showDivider: false,
                  ),
                  const SizedBox(height: 27),
                  Container(
                    width: double.infinity,
                    padding: const EdgeInsets.all(13),
                    decoration: BoxDecoration(
                      color: AppColors.infoBackground,
                      border: Border.all(color: AppColors.infoBorder),
                      borderRadius: BorderRadius.circular(12),
                    ),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          'Before your visit',
                          style: Theme.of(context).textTheme.titleMedium
                              ?.copyWith(color: AppColors.primary),
                        ),
                        const SizedBox(height: 8),
                        Text(
                          appointment.preparationNote,
                          style: Theme.of(context).textTheme.bodyLarge,
                        ),
                      ],
                    ),
                  ),
                  const SizedBox(height: 24),
                  PrimaryButton(
                    label: 'Get directions',
                    onPressed: () => _showPrototypeMessage(
                      context,
                      'Directions are not available in this prototype.',
                    ),
                  ),
                  const SizedBox(height: 8),
                  SizedBox(
                    width: double.infinity,
                    height: 53,
                    child: OutlinedButton(
                      onPressed: () => _showPrototypeMessage(
                        context,
                        'Rescheduling is not available in this prototype.',
                      ),
                      style: OutlinedButton.styleFrom(
                        foregroundColor: AppColors.primary,
                        side: const BorderSide(
                          color: AppColors.primary,
                          width: 2,
                        ),
                        shape: RoundedRectangleBorder(
                          borderRadius: BorderRadius.circular(11),
                        ),
                      ),
                      child: const Text('Reschedule'),
                    ),
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    ),
  );
}

void _showPrototypeMessage(BuildContext context, String message) {
  ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(message)));
}

class _DetailSection extends StatelessWidget {
  const _DetailSection({
    required this.label,
    required this.value,
    this.showDivider = true,
  });

  final String label;
  final String value;
  final bool showDivider;

  @override
  Widget build(BuildContext context) => Container(
    width: double.infinity,
    padding: const EdgeInsets.only(bottom: 25),
    margin: const EdgeInsets.only(bottom: 18),
    decoration: BoxDecoration(
      border: showDivider
          ? const Border(bottom: BorderSide(color: AppColors.border))
          : null,
    ),
    child: Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: Theme.of(context).textTheme.titleMedium
              ?.copyWith(color: AppColors.mutedInk),
        ),
        const SizedBox(height: 9),
        Text(value, style: Theme.of(context).textTheme.bodyLarge),
      ],
    ),
  );
}

File: react-native/src/screens/AppointmentDetailScreen.js
import { Pressable, ScrollView, StyleSheet, Text, View } from 'react-native';
import { PrimaryButton } from '../components/PrimaryButton';
import { usePrototypeFeedback } from '../components/PrototypeFeedback';
import { SafeAreaScreen } from '../components/SafeAreaScreen';
import { appointmentDetailDate } from '../utils/appointmentFormat';
import { scaledFontSize } from '../theme/tokens';
import { useClearViewTheme } from '../theme/useClearViewTheme';

export function AppointmentDetailScreen({ appointment, onBack }) {
  const { theme } = useClearViewTheme();
  const { showPrototypeFeedback } = usePrototypeFeedback();
  const location = appointment.locationDetail
    ? `${appointment.location} • ${appointment.locationDetail}`
    : appointment.location;

  return (
    <SafeAreaScreen style={{ backgroundColor: theme.colors.background }}>
      <ScrollView contentContainerStyle={styles.content}>
        <View style={styles.header}>
          <Pressable
            accessibilityLabel="Back to appointments"
            accessibilityRole="button"
            onPress={onBack}
            style={styles.back}
          >
            <Text
              style={[
                styles.backText,
                {
                  color: theme.colors.primary,
                  fontSize: scaledFontSize(34, theme),
                },
              ]}
            >
              ‹
            </Text>
          </Pressable>
          <Text
            accessibilityRole="header"
            style={[
              styles.headerTitle,
              { color: theme.colors.ink, fontSize: scaledFontSize(24, theme) },
            ]}
          >
            Appointment Details
          </Text>
        </View>
        <Text
          style={[
            styles.status,
            {
              color: theme.colors.mintInk,
              fontSize: scaledFontSize(14, theme),
            },
          ]}
        >
          {appointment.status}
        </Text>
        <Text
          style={[
            styles.title,
            { color: theme.colors.ink, fontSize: scaledFontSize(26, theme) },
          ]}
        >
          {appointment.specialty} visit
        </Text>
        <Text
          style={[
            styles.clinician,
            { color: theme.colors.ink, fontSize: scaledFontSize(18, theme) },
          ]}
        >
          {appointment.clinicianName}
        </Text>
        <Section
          label="Date & time"
          theme={theme}
          value={appointmentDetailDate(appointment.scheduledAt)}
        />
        <Section label="Location" theme={theme} value={location} />
        <Section
          label="Visit type"
          theme={theme}
          value={appointment.visitType}
        />
        <View
          style={[
            styles.notice,
            {
              backgroundColor: theme.colors.infoBackground,
              borderColor: theme.colors.infoBorder,
              borderWidth: theme.borderWidth,
            },
          ]}
        >
          <Text
            style={[
              styles.noticeTitle,
              {
                color: theme.colors.primary,
                fontSize: scaledFontSize(16, theme),
              },
            ]}
          >
            Before your visit
          </Text>
          <Text
            style={{
              color: theme.colors.ink,
              fontSize: scaledFontSize(16, theme),
            }}
          >
            {appointment.preparationNote}
          </Text>
        </View>
        <PrimaryButton
          label="Get directions"
          onPress={() =>
            showPrototypeFeedback(
              'Directions are not available in this prototype.',
            )
          }
        />
        <Pressable
          accessibilityRole="button"
          onPress={() =>
            showPrototypeFeedback(
              'Rescheduling is not available in this prototype.',
            )
          }
          style={[
            styles.outline,
            {
              borderColor: theme.colors.primary,
              borderWidth: theme.borderWidth,
            },
          ]}
        >
          <Text
            style={[
              styles.outlineText,
              {
                color: theme.colors.primary,
                fontSize: scaledFontSize(16, theme),
              },
            ]}
          >
            Reschedule
          </Text>
        </Pressable>
      </ScrollView>
    </SafeAreaScreen>
  );
}

function Section({ label, value, theme }) {
  return (
    <View
      style={[
        styles.section,
        {
          borderColor: theme.colors.border,
          borderBottomWidth: theme.borderWidth,
        },
      ]}
    >
      <Text
        style={[
          styles.sectionLabel,
          { color: theme.colors.mutedInk, fontSize: scaledFontSize(14, theme) },
        ]}
      >
        {label}
      </Text>
      <Text
        style={[
          styles.sectionValue,
          { color: theme.colors.ink, fontSize: scaledFontSize(16, theme) },
        ]}
      >
        {value}
      </Text>
    </View>
  );
}
const styles = StyleSheet.create({
  content: {
    padding: 18,
  },
  header: {
    alignItems: 'center',
    flexDirection: 'row',
    marginBottom: 20,
  },
  back: {
    justifyContent: 'center',
    minHeight: 48,
    minWidth: 48,
  },
  headerTitle: {
    fontWeight: '700',
  },
  status: {
    fontWeight: '700',
  },
  title: {
    fontWeight: '700',
    marginTop: 16,
  },
  clinician: {
    fontWeight: '700',
    marginTop: 8,
  },
  section: {
    marginTop: 24,
    paddingBottom: 18,
  },
  sectionLabel: {
    fontWeight: '700',
  },
  sectionValue: {
    marginTop: 8,
  },
  notice: {
    borderRadius: 12,
    marginVertical: 24,
    padding: 14,
  },
  noticeTitle: {
    fontWeight: '700',
    marginBottom: 8,
  },
  outline: {
    alignItems: 'center',
    borderRadius: 11,
    justifyContent: 'center',
    marginTop: 8,
    minHeight: 52,
  },
  outlineText: {
    fontWeight: '700',
  },
});
